### ISSUE TITLE
C backend won't be built on most platforms due to glibc version check

### ISSUE THREAD

**Issue by JayH5** [agronholm/cbor2#88]
Impacted: v5.1.0, master@5e8491f

cbor2 does a check that a new enough glibc is installed before enabling the C backend. It does this using the value returned by `platform.libc_ver()`. Unfortunately, this value can't be relied upon  on many versions of Python (seems like anything less than Python 3.8 because 3.8 has [bpo-35389](https://bugs.python.org/issue35389)).

For example, on Python 3.7:
```
❯ docker run --rm python:3.7-slim python -c 'import platform; print(platform.libc_ver())'
('glibc', '2.2.5')
```
...this is less than the version cbor2 requires (2.9) but the OS actually has glibc 2.28:
```
❯ docker run --rm python:3.7-slim ldd --version | head -n 1
ldd (Debian GLIBC 2.28-10) 2.28
```

Note that pip implements its own version of `libc_ver()` to get around this: https://github.com/pypa/pip/blob/20.1.1/src/pip/_internal/utils/glibc.py#L70-L87

**Comment by Sekenre** [comment]
That pip glibc library suggestion is super helpful thanks!

Would it be dangerous to depend on pip for building the C library I wonder?

I want to keep it building on Windows and Alpine too.

**Comment by JayH5** [comment]
> Would it be dangerous to depend on pip for building the C library I wonder?

I would have to say yes, it would be dangerous 😬. It is/has been (in my experience) pretty common for libraries to depend on pip internals and then break when there's a new pip release. See here: https://pip.pypa.io/en/stable/user_guide/#using-pip-from-your-program

I'm not really sure how other libraries handle this.. I may have a look around for examples if I have time.

One thing maybe worth considering is whether the version check is necessary at all given that glibc 2.9 was released back in 2009. In the Python space, one reference to glibc versions is the manylinux standards:
* PEP-513 `manylinux1`: glibc 2.5
* PEP-571 `manylinux2010`: glibc 2.12
* PEP-599 `manylinux2014`: glibc 2.17

cbor2 would currently only not support `manylinux1` but it is (sort of) obsolete as it's based on CentOS 5 which is no longer maintained. These are just packaging standards, though, not runtime requirements for Python itself.

**Comment by Sekenre** [comment]
Thanks, I agree. I will borrow from the pip example.

This was written to fix #67 but it looks like a better option is available.

**Comment by Sekenre** [comment]
Hi @JayH5 

Can you try this:
```
pip install https://github.com/Sekenre/cbor2.git@fix_glibc_compat#egg=cbor2
```

It should fix the problem

**Comment by JayH5** [comment]
Yep, looks good, thanks 👍 
```
❯ docker run --rm -it python:3.7 bash
root@d5aedded56ee:/# ldd --version | head -n 1
ldd (Debian GLIBC 2.28-10) 2.28
root@d5aedded56ee:/# pip install git+https://github.com/Sekenre/cbor2.git@fix_glibc_compat#egg=cbor2
Collecting cbor2
  Cloning https://github.com/Sekenre/cbor2.git (to revision fix_glibc_compat) to /tmp/pip-install-_1ibgncz/cbor2
  Running command git clone -q https://github.com/Sekenre/cbor2.git /tmp/pip-install-_1ibgncz/cbor2
  Running command git checkout -b fix_glibc_compat --track origin/fix_glibc_compat
  Switched to a new branch 'fix_glibc_compat'
  Branch 'fix_glibc_compat' set up to track remote branch 'fix_glibc_compat' from 'origin'.
Building wheels for collected packages: cbor2
  Building wheel for cbor2 (setup.py) ... done
  Created wheel for cbor2: filename=cbor2-5.0.0.post27-cp37-cp37m-linux_x86_64.whl size=188292 sha256=fe91b1a02da3c9476acb1aa2019a4b000ba32fc5a3b00f3f3a0e204e63ae1797
  Stored in directory: /tmp/pip-ephem-wheel-cache-dejju803/wheels/95/62/d9/8c460e65528edc9e118ee3218d59bbb1bf8a63ec4447248a15
Successfully built cbor2
Installing collected packages: cbor2
Successfully installed cbor2-5.0.0.post27
root@d5aedded56ee:/# ldd /usr/local/lib/python3.7/site-packages/_cbor2.cpython-37m-x86_64-linux-gnu.so
	linux-vdso.so.1 (0x00007ffe027e6000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fb3dc644000)
	libpython3.7m.so.1.0 => /usr/local/lib/libpython3.7m.so.1.0 (0x00007fb3dc2e3000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fb3dc2c2000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fb3dc101000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fb3dc7f0000)
	libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007fb3dc0c7000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fb3dc0c2000)
	libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007fb3dc0bb000)
```

**Comment by Sekenre** [comment]
Fix released in 5.1.1

Thanks for your help!
